shared/middleware/helpers/storage_apps.rb

```diff
@@ -264,6 +264,33 @@ def project_type_from_channel_id(channel_id)
     project_type_from_merged_row(get(channel_id))
   end
 
+  #
+  # Looks up the set of ancestors in the remix history for a particular project.
+  # This can require several queries, so be careful exposing this in the UI
+  # for external users - right now this is designed as a utility for internal
+  # use only.
```

Comment:
It should be possible to reduce the number of queries by joining the table against itself. No need to do this now, but perhaps worth mentioning here as a solution to the performance problem if we expose this to users.